### DISCUSSION TITLE
How to enable full brightness on a CCT-LED-Strip?

### DISCUSSION THREAD

**Discussion by bugfixer1992** [arendst/Tasmota#18243] [Q&A]
Dear Tasmota-Community,
I try to control my CCT-LED-Strip (https://www.led-konzept.de/CCT-LED-Streifen-24V-dualweiss-22W-m-warmweiss-kaltweiss) with a Shelly RGBW2. Since the stock firmware left a lot to be desired I flashed Tasmota, configured it and everything works as expected.
When I set the color temperature to be in the middle of the spectrum and set the brightness to 100% Tasmota sets the channels to [50,50], which basically means that both channels make use of 50% of their maximum brightness. This absolutely makes sense in cases, where you are looking for a constant level of brightness, or your power-supply cannot handle both channels to be at maximum brightness.
Since I made sure that the power supply and heat-dissipation are fine with both channels on maximum brightness I am looking for an additive, not linear CCT blending approach like shown in https://kno.wled.ge/features/cct/#:~:text=and%20realtime%20sources.-,CCT%20additive%20blending,-Setting%20this%20to. I am using the strip mostly in and around the middle of the color spectrum and so I don´t care about even, but maximum brightness. By the way, if I set both channels manually to 100 with “Backlog Channel1 100; Channel2 100” the LED-Strip gets significantly brighter. 
I already looked into the code and I guess that somewhere in xdrv_04_light.ino there is the logic implemented. I even tinkered with the “free_range” but without look so far. I definitely have no problem with low level coding in general or compiling the custom firmware, so even a solution where the code has to be changed would be okay.
But I think that this could be a nice feature to be implemented for many other people. To have the possibility to change the CCT additive blending directly in the config. Ideally there would be a input from 0-100% to achieve the different outputs shown in https://kno.wled.ge/features/cct/#:~:text=and%20realtime%20sources.-,CCT%20additive%20blending,-Setting%20this%20to and everything in between.
Does someone of you know how to achieve this?
Any help would be highly appreciated 😊
Thanks and kind regards, 
Philipp

**Comment by bugfixer1992** [comment]
@s-hadinger, do you have any idea how to achieve this?

**Reply by s-hadinger** [reply]
Yes it's possible but made difficult to do on purpose, to avoid accidental destruction of devices if the power supply does not support all channels to be at maximum level at the same time.

Just use `Color FFFF`. When specifying levels via the `Color` command, there is no limitation in overall power.

**Reply by bugfixer1992** [reply]
Thank you so much for your reply! 

I experienced that manually setting the color to FFFF has the same effect than doing "Backlog Channel1 100; Channel2 100". 
But when I change the color temperature even a tiny bit the brightness drops back to the old level. 
Is there any possibility to always get the maximum brigthtness, no matter what input or function I use?

**Reply by s-hadinger** [reply]
No. It is really a safety measure to avoid destruction of devices

**Reply by bugfixer1992** [reply]
Okay, I understand. But wouldn´t it be a good compromise to have a blending-feature like the one implemented in WLED?
Basically a parameter you can set between 0 and 100 to estimate how much blending of both channels you would like to have. 
There is an example in https://kno.wled.ge/features/cct/. If there is some interest in implementing such a feature I would be glad to help.

![image](https://user-images.githubusercontent.com/54179038/229610300-b99d102b-46a3-4a28-9e56-d5880aa85050.png)

**Reply by s-hadinger** [reply]
Actually it's much more complex because you need to take into account gamma correction into this. This is really an edge case. The light code is already complex enough, I'm not willing to add this and make it even more complex

**Reply by jrm523** [reply]
That's unfortunate. I too am in a similar situation.  Regardless, thanks for what you do.

**Reply by sfromis** [reply]
Basically, 50% of each is the "real" full brightness, otherwise you'd see the brightness fall when moving towards all cold or all warm. And much in commercial hardware is designed to work this way, with constant power when varying CT across a certain brightness.

**Reply by barbudor** [reply]
You can use separated PWM channels and control the way you like

**Comment by rutkas** [comment]
Regarding this function on WLED. Version 0.15.1 still does not save CCT additive blending permanent, but applies it until reboot. However you can save any value permanently by exporting config, setting “cb” value to 100 or what you need and importing config back to controller. After that new default value is set and applied after reboot.
